src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,59 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
 public class SalaryInfo {
+    private static final DateTimeFormatter DATEFORMAT = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+
+        LocalDate fromDate;
+        LocalDate toDate;
+        try {
+            fromDate = LocalDate.parse(dateFrom, DATEFORMAT);
+            toDate = LocalDate.parse(dateTo, DATEFORMAT);
+        } catch (DateTimeParseException e) {
+            return "Invalid date format";
+        }
+
+        int[] salaries = new int[names.length];
+
+        for (String entry : data) {
+            String[] entryParts = entry.split(" ");
+            if (entryParts.length == 4) {
+                LocalDate entryDate = LocalDate.parse(entryParts[0], DATEFORMAT);
```

Comment:
add constants for indexes

```suggestion
                LocalDate entryDate = LocalDate.parse(entryParts[DATE_INDEX], DATEFORMAT);
```